A python script to periodically collect states

themarwhal opened the pull request:
Differential Revision: D14754977

shahrahman-fb commented:
Hi, I just noticed that the title appears to be different than the actual contents of this PR.

Is that intentional?
________________________________
From: Marie <notifications@github.com>
Sent: Thursday, April 4, 2019 10:14 AM
To: facebookincubator/magma
Cc: Subscribed
Subject: [facebookincubator/magma] A python script to periodically collect states (#167)


Differential Revision: D14754977

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/facebookincubator/magma/pull/167

Commit Summary

  *   Implement an in-memory blobstore
  *   Setting up servicers for the state service
  *   Adding a GetOperationalStates method
  *   A python script to periodically collect states

File Changes

  *   M orc8r/cloud/configs/service_registry.yml<https://github.com/facebookincubator/magma/pull/167/files#diff-0> (5)
  *   A orc8r/cloud/configs/state.yml<https://github.com/facebookincubator/magma/pull/167/files#diff-1> (7)
  *   M orc8r/cloud/deploy/files/envdir/CONTROLLER_SERVICES<https://github.com/facebookincubator/magma/pull/167/files#diff-2> (2)
  *   M orc8r/cloud/deploy/roles/controller/files/supervisord.conf<https://github.com/facebookincubator/magma/pull/167/files#diff-3> (7)
  *   A orc8r/cloud/go/blobstore/memory.go<https://github.com/facebookincubator/magma/pull/167/files#diff-4> (335)
  *   A orc8r/cloud/go/blobstore/memory_test.go<https://github.com/facebookincubator/magma/pull/167/files#diff-5> (396)
  *   M orc8r/cloud/go/defs.mk<https://github.com/facebookincubator/magma/pull/167/files#diff-6> (2)
  *   M orc8r/cloud/go/pluginimpl/serdes.go<https://github.com/facebookincubator/magma/pull/167/files#diff-7> (3)
  *   M orc8r/cloud/go/protos/service303.pb.go<https://github.com/facebookincubator/magma/pull/167/files#diff-8> (274)
  *   A orc8r/cloud/go/protos/state.pb.go<https://github.com/facebookincubator/magma/pull/167/files#diff-9> (419)
  *   M orc8r/cloud/go/service/service303.go<https://github.com/facebookincubator/magma/pull/167/files#diff-10> (6)
  *   A orc8r/cloud/go/services/state/client_api.go<https://github.com/facebookincubator/magma/pull/167/files#diff-11> (132)
  *   A orc8r/cloud/go/services/state/client_api_test.go<https://github.com/facebookincubator/magma/pull/167/files#diff-12> (77)
  *   A orc8r/cloud/go/services/state/servicers/proto_validation.go<https://github.com/facebookincubator/magma/pull/167/files#diff-13> (48)
  *   A orc8r/cloud/go/services/state/servicers/servicer.go<https://github.com/facebookincubator/magma/pull/167/files#diff-14> (112)
  *   A orc8r/cloud/go/services/state/state/main.go<https://github.com/facebookincubator/magma/pull/167/files#diff-15> (43)
  *   A orc8r/cloud/go/services/state/test_init/test_service_init.go<https://github.com/facebookincubator/magma/pull/167/files#diff-16> (32)
  *   M orc8r/gateway/c/common/service303/MagmaService.cpp<https://github.com/facebookincubator/magma/pull/167/files#diff-17> (8)
  *   M orc8r/gateway/c/common/service303/MagmaService.h<https://github.com/facebookincubator/magma/pull/167/files#diff-18> (13)
  *   M orc8r/gateway/python/magma/common/service.py<https://github.com/facebookincubator/magma/pull/167/files#diff-19> (19)
  *   M orc8r/gateway/python/magma/magmad/checkin_manager.py<https://github.com/facebookincubator/magma/pull/167/files#diff-20> (1)
  *   M orc8r/gateway/python/magma/magmad/main.py<https://github.com/facebookincubator/magma/pull/167/files#diff-21> (5)
  *   A orc8r/gateway/python/magma/magmad/state_manager.py<https://github.com/facebookincubator/magma/pull/167/files#diff-22> (110)
  *   M orc8r/protos/service303.proto<https://github.com/facebookincubator/magma/pull/167/files#diff-23> (12)
  *   A orc8r/protos/state.proto<https://github.com/facebookincubator/magma/pull/167/files#diff-24> (43)

Patch Links:

  *   https://github.com/facebookincubator/magma/pull/167.patch
  *   https://github.com/facebookincubator/magma/pull/167.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/facebookincubator/magma/pull/167>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AtVlL2ImEid3fKtkuPMvKdVJO_NNOf-5ks5vdjLxgaJpZM4cdTO1>.

themarwhal committed:
Add logic to periodically collect operational states from gateway services (#167)

Summary:
Pull Request resolved: https://github.com/facebookincubator/magma/pull/167

A new service303 function GetOperationalStates was added in D14738655. A new script in magmad, state_reporter, will query each state for its operational states, aggregate them, and send them off to the state service. Since the operational states can look different for every device, it is encoded as a JSON.

Each service can register a callback function to write in the state when GetOperationalStates is called by the state_reporter. The default in every service is None. This is the same model that the service_poller uses.

Additionally, the reporter will also take the latest checkin request sent as a state and report it.

Reviewed By: xjtian

Differential Revision: D14754977

fbshipit-source-id: af01a62b881f88e589fdb0a87d243e49cc8c55c1
